### DISCUSSION TITLE
Ride Hailing Dicussion for Rental/Schedule/outstation

### DISCUSSION THREAD

**Discussion by 92shreyansh** [ONDC-Official/mobility-specification#158] [Ideas]
Hi All,

Please use this space of discussion to ideate on ride hailing use cases below:

- Schedule
- Rental
- Outstation

**Comment by shobhanaquickride** [comment]
IMO, on-demand and scheduled should be made available for all ride types - Local (includes Airport as well), Rental and Outstation. The only difference between on-demand and scheduled rides would be how much time in advance is the booking made. So we should have on-demand local rides, scheduled local rides, on-demand rental rides, scheduled rental rides and on-demand outstation rides, scheduled outstation rides.

For both on-demand and scheduled rides of all kinds, the driver and vehicle information would be shared in Async mode.

What differentiates between Local, Rental and Outstation rides? 
1. Local rides are point-to-point rides within the same city. 
2. Rental rides are open ended rides with just the start location and can end anywhere within the same city. 
3. Outstation rides are point-to-point inter-city rides which can be one-way or multi-way.

**Comment by shobhanaquickride** [comment]
BRD says "Buyer receives driver and vehicle information immediately for on-demand rides and before a pre-determined duration for scheduled rentals (120 mins or 90 mins)"

This should not be mandated by ONDC but left to the seller apps. As part of product discovery, seller app can specify how much time in advance the driver and vehicle information would be shared with the buyer

**Reply by wridelife** [reply]
To keep a well performing network and have each party adhere to a certain rule, recommendation is to  keep the minimum time by when Vehicle and Driver information is shared. As an example, at least 20 minutes prior to the ride start. this will make sure that everyone adheres to the minimum.

**Reply by anurag-singh-12** [reply]
I have tried to explain the need for an ONDC mandated cut-off time in my comment below(advances and cancellations). While seller apps can have 1:1 understanding with individual buyer apps on the cut-off to share the driver details (which can be better than the network mandated cut-off), but the need of advances and cancellation fees is paramount in long distance travel because the cost of cancellation is very high for the driver. This can only be regulated if there is a network mandated cut-off, which has to be followed by everyone instead of 1:1 understanding between seller and buyer apps. 
Additionally, it will become very difficult to retain drivers on the seller network without the cancellation fees or the cost of cancellation will have to borne by seller apps.

**Comment by wridelife** [comment]
Cancellation is a major concern for both passengers as well as drivers and a big hassle to NPs, if NPs do not integrate and mandate credit card, then it becomes a major concern for NPs.
Whether on demand or Scheduled, there needs to be a partial or full payment provision where the money stays neutral to Passenger or driver and in case of breach of cancellation rules a penalty can be levied on the party breaking the rule and proper compensation be provided to the affected party.

**Comment by anurag-singh-12** [comment]
**Advance:**
All sorts of long distance travel (intercity, outstation & rentals) should collect advance from the buyers/customers.  Cab drivers don’t have access to a lot of working capital and will require the advance to pay for the expenses (fuel, recharge of fast tags etc) accrued in long distance travel

**Comment by anurag-singh-12** [comment]
**Cancellations:**
Further, there is a need to charge cancellation fees to the customer in case of last minute cancellation (ideally, beyond the cutoff time mandated by ONDC or fixed with the buyer app individually) because:
1. Scenario 1: Driver has planned the travel from point A to point B and back to point A, if the ride from point A to B is cancelled at the last moment , then the follow-up ride from point B to point A will also be impacted.
2. Driver’s base location is point A, and has a trip planned from point A to point B. He accepts a trip (at low fare) from point B to point A to complete the round trip. If the second half of the trip is cancelled, that is point B to point A then the driver might have to spend a significant amount of time waiting at the at point B to return back to his base location, which will be additional cost for the driver.

**Comment by shobhanaquickride** [comment]
As part of discovery, we should have option to specify the fuel type. This can help the buyer to estimate the luggage capacity as well.
